### ISSUE TITLE
Support for Jackson 2.12

### ISSUE THREAD

**Issue by hisener** [mongojack/mongojack#214]
Since https://github.com/mongojack/mongojack/pull/194, `DBDecoderBsonParser` is no longer a `de.undercouch.bson4jackson.BsonParser`. As a result, after the MongoJack upgrade, we hit the same issue in https://github.com/michel-kraemer/bson4jackson/issues/93.

Does it make sense to override `#getNumberValueExact` method and delegate to `#getNumberValue` like `BsonParser`?

**Workaround**

```java
private static final class Jackson212CompatibleDBDecoderBsonParser extends DBDecoderBsonParser {
    private Jackson212CompatibleDBDecoderBsonParser(
            IOContext ctxt,
            int jsonFeatures,
            AbstractBsonReader reader,
            ObjectMapper objectMapper,
            UuidRepresentation uuidRepresentation) {
        super(ctxt, jsonFeatures, reader, objectMapper, uuidRepresentation);
    }

    @Override
    public Number getNumberValueExact() {
        return getNumberValue();
    }
}
```

**Comment by rlodge** [comment]
This is on my list of things to look at, but I'm swamped this week.

**Comment by hisener** [comment]
Thanks @rlodge! I've created https://github.com/mongojack/mongojack/pull/215. All tests are passed but not sure if there is anything else we need to check.

**Comment by rlodge** [comment]
I added some comments to the pull request.  No significant issues, but if you could respond that would be great.

**Comment by rlodge** [comment]
I released a version 4.2.1 with this today.
